File: qc-domain/src/main/java/com/eager/questioncloud/domain/question/repository/QuestionCategoryRepository.java
package com.eager.questioncloud.domain.question.repository;

import com.eager.questioncloud.domain.question.dto.QuestionCategoryGroupBySubject.MainQuestionCategory;
import java.util.List;

public interface QuestionCategoryRepository {
    List<MainQuestionCategory> getMainQuestionCategories();
}

File: qc-core/src/main/java/com/eager/questioncloud/core/domain/question/infrastructure/repository/QuestionCategoryRepository.kt
package com.eager.questioncloud.core.domain.question.infrastructure.repository;

import com.eager.questioncloud.core.domain.question.dto.QuestionCategoryGroupBySubject.MainQuestionCategory;
import java.util.List;

public interface QuestionCategoryRepository {
    List<MainQuestionCategory> getMainQuestionCategories();
}
